esphome/dashboard/dashboard.py

```diff
@@ -927,6 +927,25 @@ def get(self):
         self.write(json.dumps(secret_keys))
 
 
+class JsonConfigRequestHandler(BaseHandler):
+    @authenticated
+    @bind_config
+    def get(self, configuration=None):
+        filename = settings.rel_path(configuration)
+        if not os.path.isfile(filename):
+            self.send_error(404)
+            return
+
+        try:
+            content = yaml_util.load_yaml(filename, clear_secrets=False)
+            json_content = json.dumps(content, default=lambda o: f"{repr(o)}")
```

Comment:
What is the default keyword doing here. What objects can it not serialize?

Reply:
The one's that I've found during testing `!lambda` blocks don't serialize.

Reply:
I suggest we go with a more compatible type, that can be extended in the future. 
```suggestion
            json_content = json.dumps(content, default=lambda o: {"__type": str(type(o)), "repr": repr(o)})
```

([copied from HA](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/json.py#L55-L60))